pkg/client/outofband/client.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package outofband
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/google/uuid"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/decorator"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+)
+
+const (
+	protocolURI    = "https://didcomm.org/oob-request/1.0"
+	requestMsgType = protocolURI + "/request"
+)
+
+// RequestOptions allow you to customize the way request messages are built.
+type RequestOptions func(*Request) error
+
+// ConnectionRecorder records connection records produced as byproducts of creation of messages.
+type ConnectionRecorder interface {
+	SaveInvitation(id string, i interface{}) error
+}
+
+// Provider provides the dependencies for the client.
+type Provider interface {
```

Comment:
Does context fulfills this provider ? Usually [framework.Context()](https://github.com/hyperledger/aries-framework-go/blob/23fd9909a22991088bf646e981e479add5e1bc31/pkg/client/didexchange/client.go#L34) is used to create the client.

Reply:
Does it need to? Is there a reason why it can't be adapted in eg. `pkg/framework/aries/default.go`?

Reply:
Yeah, usually we add the provider dependency in the context and the consumer can use the context object directly to initialize the client.

Reply:
tagging - @troyronda @sudeshrshetty @fqutishat

Reply:
What if someone wants to customize it?  only entry in defaults won't be enough.
You don't need to connection recorder in provider, instead you should use store providers here.
You can refer didexchange service to use `connection.NewRecorder(ctx)`

Reply:
I opened #1467

Can this PR be merged?

Reply:
connectionRecorder never intended to be a interface

Reply:
It is an interface that this package declares as a dependency.

Reply:
@llorllale do you have a BDD test of this package?

I'm not quite understanding the usage plan when the framework generated context cannot be used?

Reply:
Also - I would think the constructor would take care of details of consuming the framework context, rather than requiring an adapter utility function.

Reply:
@troyronda BDD tests TODO

> I'm not quite understanding the usage plan when the framework generated context cannot be used?

I plan on addressing the "context" problem in #1467. Does that answer your question?